lib/task/import/importDipObjectsTask.class.php

```diff
@@ -0,0 +1,585 @@
+   *
+   * @return void
+   */
+  protected function addDigitalObject($informationObject, $filepath, $undoLog = null, $container = false)
+  {
+    // Abort if a digital object already exists for this information object
+    if (null !== $informationObject->getDigitalObject())
+    {
+      $this->log("A digital object is already attached to $informationObject->identifier (slug: $informationObject->slug). Skipping.");
+
+      return;
+    }
+
+    // Make sure file exists
+    if (!file_exists($filepath))
+    {
+      throw new Exception("Couldn't find file '$filepath'");
+    }
+
+    // Create digital object
+    $this->log("Importing '$filepath'");
+
+    $do = new QubitDigitalObject;
+    $do->usageId = QubitTerm::MASTER_ID;
+    $do->assets[] = new QubitAsset($filepath);
+
+    // Add digital object to information object
+    $informationObject->digitalObjects[] = $do;
+
+    // Add DIP UUID as aipUUID information object property
+    if (null !== $dipUUID = $this->getUUID(basename($this->dipDir)))
+    {
+      $this->log('Creating property: dip UUID '. $dipUUID);
+      $informationObject->addProperty('aipUUID', $dipUUID);
+    }
+
+    // Add object UUID as objectUUID information object property
+    if (null !== $objectUUID = $this->getUUID(basename($filepath)))
+    {
+      $this->log('Creating property: object UUID '. $objectUUID);
+      $informationObject->addProperty('objectUUID', $objectUUID);
+    }
+
+    // Save and, optionally, log
+    $informationObject->save($this->conn);
+
+    if (isset($undoLog))
+    {
+      $logLine = $informationObject->id ."\t". basename($this->dipDir) ."\t". $container ."\n";
+      file_put_contents($undoLog, $logLine, FILE_APPEND);
+    }
+  }
+
+  /**
+   * Get upload directory from settings
+   *
+   * @return string  Path to upload directory
+   */
+  protected function getUploadDir()
+  {
+    $uploadDir = 'uploads'; // Default value
+
+    $sql = 'SELECT i18n.value
+      FROM setting stg JOIN setting_i18n i18n ON stg.id = i18n.id
+      WHERE stg.source_culture = i18n.culture
+       AND stg.name = \'upload_dir\';';
+
+    if ($sth = $this->conn->query($sql))
+    {
+      list($uploadDir) = $sth->fetch();
+    }
+
+    return $uploadDir;
+  }
+
+  /**
+   * Create a lookup table that will be used to match original filename
+   * (example: foo.TIF) with its DIP object filename (example:
+   * 979c4458-21f3-11e1-a4bd-001d09282b9d-foo.jpg)
+   *
+   * The lookup table stores files by their basename in this format:
+   *
+   * Array
+   * (
+   *   [somefile] => ad0acf68-34b4-47ea-aef7-e5e29c5b7388-somefile.tif
+   * )
+   *
+   * @param string $objectsPath  Path to DIP objects
+   *
+   * @return array  Hash (see function description)
+   */
+  protected function createFilenameLookup($objectsPath)
+  {
+    if (!($files = scandir($objectsPath)))
+    {
+      throw new sfException("Couldn't read path '$objectsPath'");
```

Comment:
I think we could remove this check, at this point we should already have the csv file from the objects directory.